src/main/java/games/strategy/engine/lobby/server/db/UserController.java

```diff
@@ -29,19 +30,30 @@ public UserController() {
    * @return null if the user does not exist.
    */
   @Override
-  public HashedPassword getPassword(final String userName) {
-    final String sql = "select password from ta_users where username = ?";
-    try (final Connection con = connectionSupplier.get(); final PreparedStatement ps = con.prepareStatement(sql)) {
-      ps.setString(1, userName);
+  public HashedPassword getPassword(final String username) {
+    return getPasswordFromColumn(username, "coalesce(password, bcrypt_password)");
+  }
+
+  /**
+   * @return null if the user does not exist or the user has no legacy password stored.
+   */
+  @Override
+  public HashedPassword getMd5Password(final String username) {
+    return getPasswordFromColumn(username, "password");
+  }
+
+  private HashedPassword getPasswordFromColumn(final String username, final String column) {
+    try (final Connection con = connectionSupplier.get();
+        final PreparedStatement ps = con.prepareStatement("select " + column + " from ta_users where username=?")) {
```

Comment:
In case anyone wonders: No, this can't be abused by SQLInjection attacks.

Reply:
Variable arg column is a bit of a wtf. Just write two SQL statements directly and it stays pretty simple. It also conveys that one case we are doing login, the other case for some reason we insist on getting the legacy password. DRY in this case I think is harmful.